aea/aea.py

```diff
@@ -297,11 +297,11 @@ def _handle_decoding(
             return None, []  # Tuple[Optional[Message], List[Handler]]
 
         handlers = self.filter.get_active_handlers(
-            protocol.public_id, envelope.to_as_public_id or envelope.skill_id
+            protocol.public_id, envelope.to_as_public_id
```

Comment:
For component to component we don't need the envelope context. For agent to agent, the other agent should not be able to target a specific skill.